### PR TITLE
Fix skills model

### PR DESCRIPTION
**Pull request by yx296**
This PR allows for skills to be sent to the server and fixes the isolate scope for the tagsInput directive.
It also allows for display of skills to be seen in details.
 
 I changed the tests slightly so they would pass, not sure if faker can generate a fake set of skills.

### CODE REVIEW

**Comment by yx296** [comment]
Insert a space commit passes all the tests locally but fails Travis...

**Comment by yx296** [comment]
and now travis passes after a rather inconsequential commit, wut

File: client/vivace/details/controller.js [hunk 1]
```diff
@@ -13,7 +13,12 @@ module.exports = [
       .then((data) => {
         this.user = data;
         this.user.name = data.firstName + ' ' + data.lastName;
+        this.user.skills = data.skills.map((skillObj) => {
+          return skillObj.text;
+        });
+        this.user.skillsDisplayStr = this.user.skills.join(', ');
```
Comment (jackofseattle) [inline client/vivace/details/controller.js]: Why do we need this? Seems odd to do this in here.

File: client/vivace/details/template.html [hunk 2]
```diff
@@ -30,12 +30,16 @@
       </md-input-container>
 
       <md-input-container class="md-icon-float md-block">
+        <label>Skills</label>
+        <input ng-model="details.user.skillsDisplayStr" type="text" class="short-description border-bottom-none" disabled="true">
```
Comment (jackofseattle) [inline client/vivace/details/template.html]: this is a disabled input, is there no other way to get this to work?

File: client/vivace/details/controller.js [hunk 3]
```diff
@@ -13,7 +13,12 @@ module.exports = [
       .then((data) => {
         this.user = data;
         this.user.name = data.firstName + ' ' + data.lastName;
+        this.user.skills = data.skills.map((skillObj) => {
```
Comment (jackofseattle) [inline client/vivace/details/controller.js]: one liner will work fine here.  ```data.skills.map(x => x.text);```

**Comment by yx296** [comment]
Changed map to one liner. All the other fields are disabled inputs, so I decided to use that as example for the skills display.

**Pull request merged**